docs/proposals/scheduler/agent.proto

```diff
@@ -0,0 +1,73 @@
+syntax = "proto3";
+
+message Empty {}
+
+
+// ------------------
+// Lifecycle
+// ------------------
+
+message ConnectionRequest {
+    string uuid = 1;
+}
+
+message ConnectionResponse {
+    enum StatusCode {
+        GRANTED = 0;
+        DENIED = 1;
+    }
+
+    StatusCode status_code = 1;
+}
+
+service LifecycleService {
+    rpc Connect(ConnectionRequest) returns (ConnectionResponse);
+    rpc Disconnect(Empty) returns (Empty);
+}
+
+
+// ------------------
+// Status updates
+// ------------------
+
+message NodeStatus {
+    message Memory {
+        uint64 total = 1;
+        uint64 free = 2;
+    }
+
+    message CpuLoad {
+        double load = 1;
+    }
+
+    Memory memory = 1;
+    CpuLoad cpu_load = 2;
+}
+
+message InstanceStatus {
+    enum StatusCode {
+        WAITING = 0;
+        RUNNING = 1;
+        TERMINATED = 2;
+    }
+
+    message Resources {
+        int32 cpu = 1;
+        int32 memory = 2;
+        int32 disk = 3;
+    }
+
+    string name = 1;
+    StatusCode status_code = 2;
+    Resources resource_usage = 3;
+    string message = 4;
+}
+
+message InstanceStatuses {
+    repeated InstanceStatus instance_statuses = 1;
+}
+
+service StatusUpdateService {
+    rpc UpdateNodeStatus(NodeStatus) returns (Empty);
+    rpc UpdateInstanceStatuses(InstanceStatuses) returns (Empty);
```

Comment:
You're right that this won't be necessary when the instance is created because we immediately receive a status. 
What I had more in mind was during the normal lifecycle of an instance, if the instance crashes, the `UpdateInstanceStatuses` function would have been called to inform the scheduler that this specific instance is not running anymore, which would later trigger the re-scheduling of that instance.
(Taking into account [your other comment](https://github.com/dev-sys-do/orka/pull/2#discussion_r1271786282), this should be returning a stream too)

What's your opinion on this?

Reply:
Ok just understood what your message meant, must have been still asleep this morning. I will update the proto later in that regard.